uniswap-v2/logics/impls/router/router.rs

```diff
@@ -171,10 +171,18 @@ impl<T: Storage<data::Data>> Router for T {
             liquidity,
         )?;
 
-        let (amount_0, amount_1) = PairRef::burn_builder(&pair_contract, to)
+        let (amount_0, amount_1) = match PairRef::burn_builder(&pair_contract, to)
             .call_flags(CallFlags::default().set_allow_reentry(true))
             .fire()
-            .unwrap()?;
+        {
+            Ok(res) => {
+                match res {
+                    Ok(v) => Ok(v),
+                    Err(err) => Err(RouterError::PairError(err)),
+                }
+            }
+            Err(_) => Err(RouterError::TransferError),
```

Comment:
Same as https://github.com/swanky-dapps/dex/pull/1/files#r1041638816

Reply:
For router error, we can introduce ink error as a new RouterError to wrap it; but to be consistent with above "deposit error" handling, it is good as it is.